proseco/fid.py

```diff
@@ -319,7 +334,8 @@ def get_fid_candidates(self):
         for idx, fid in enumerate(cand_fids):
             if (self.off_ccd(fid) or
                     self.near_hot_or_bad_pixel(fid) or
-                    self.has_column_spoiler(fid, self.stars, stars_mask)):
+                    self.has_column_spoiler(fid, self.stars, stars_mask) or
+                    self.is_excluded(fid)):
```

Comment:
I think that include_ids should be handled here, which is analogous to acq and guide. If a fid is force-included, then it should be in the selection no matter what, unless it is not valid.  Not-tested code:
```
        if id_diff := set(self.include_ids_fid) - set(cand_fids['id'])
            raise ValueError(f'included fid ids {id_diff} are not valid')

        for fid in cand_fids:
            excluded = (self.off_ccd(fid)
                        or self.near_hot_or_bad_pixel(fid)
                        or self.has_column_spoiler(fid, self.stars, stars_mask)
                        or self.is_excluded(fid))
            included = fid['id'] in self.include_ids_fid

            if not included and excluded:
                idx_bads.append(idx)
```

Reply:
My concern with that is that I don't think sparkles is doing near pixel or column spoiler checks on fids,  so if we allow override in this way (yes, analogous to acq/fid) to be strictly correct we'd also have to implement the checks in sparkles.  My approach of just reducing the candidates was somewhat weak but benign in this way (if no combinations existed, with the included fid and all fids were excluded, at least you'd get the "no fids" warning out of sparkles).

But I suppose we can just add the checks in sparkles (at least I'm pretty sure they aren't in there now).

Reply:
Adding those checks to sparkles (and starcheck?) would be a good thing, but not strictly required for this PR. 99.9% of the use for forcing fids will be by starcheck, and there you really need to force the exact same catalog as in loads.

Reply:
I don't have any walrus operator experience, but that first test seems not to work.   Does it need to be explicitly checked == 0 or what is your preference?

Reply:
Never mind.  Seems to work now.